### PR TITLE
Bug 1495460 - Retry describe-clusters request on failure

### PR DESCRIPTION
**Pull request by jklukas**
This PR also includes some other fixes from our early experience with running clusters that have metrics turned on:

- add some extra quoting on our tags in the YAML configuration for datadog
- restrict the scope of metrics we're sending

### CODE REVIEW

File: ansible/files/bootstrap/telemetry.sh [hunk 1]
```diff
@@ -270,10 +285,10 @@ EOF
     sudo tee -a /etc/datadog-agent/datadog.yaml <<EOF
 
 tags:
-- cluster_name:${CLUSTER_NAME}
-- cluster_id:${CLUSTER_ID}
+- cluster-name:"${CLUSTER_NAME}"
+- cluster-id:"${CLUSTER_ID}"
```
Comment (jklukas) [inline ansible/files/bootstrap/telemetry.sh]: Datadog seems to consistently use kebab-case in its metrics naming, so I'm moving to that here.

**Comment by jklukas** [comment]
I was able to successfully spin up a cluster with these changes, so the retry logic works at least on the happy path. The decryption content in S3 doesn't exist for dev, though, so it's impossible to test that the yaml changes work as expected.

**Comment by jklukas** [comment]
Added some permissions fixes.

File: ansible/files/bootstrap/telemetry.sh [hunk 2]
```diff
@@ -14,14 +14,29 @@ then
...
+        let "sleep_seconds = sleep_seconds * 2"
+    done
+}
 describe_cluster() {
     local query=$1
     jq -r "$query" /mnt/describe-cluster.json
 }
 
 # Pull information about the cluster.
-CLUSTER_ID=$(jq -r .jobFlowId /mnt/var/lib/info/job-flow.json)
```
Comment (sunahsuh) [inline ansible/files/bootstrap/telemetry.sh]: nit: would be nice to keep this global assignment out of the function (as far as I can tell there's no particular advantage in moving it to the function?)

**Pull request merged**